datafusion/physical-expr/src/planner.rs

```diff
@@ -83,6 +83,84 @@ pub fn create_physical_expr(
                 }
             }
         }
+        Expr::IsTrue(expr) => {
+            let binary_op = binary_expr(
+                expr.as_ref().clone(),
+                Operator::IsNotDistinctFrom,
+                Expr::Literal(ScalarValue::Boolean(Some(true))),
```

Comment:
Cool!

One thing is that when we pass in something other than a boolean (like a number) with something like `SELECT 2 IS TRUE as t`, we get an error that looks like `Error: Plan("'Int64 IS NOT DISTINCT FROM Boolean' can't be evaluated because there isn't a common type to coerce the types to")`. While this generally looks like the desired error, the `IS NOT DISTINCT FROM` might be a little confusing from the user's POV.

Reply:
Would the assumption be that any number that is not `1` would result in False in this case?

Reply:
Now that we have the specific `IsTrue` expressions in the logical plan, we can update the `TypeCoercion` optimizer rule to validate (or try and coerce) the types before we get to the physical planning phase. I have filed https://github.com/apache/arrow-datafusion/issues/3304 to track this.

Note that for the Dask SQL usage, we will no longer see any references to `IsNotDistinctFrom` since that mapping is happening in DataFusion's physical query planner, which we are not using.

Reply:
@jdye64 @sarahyurick  
PG just support boolean data type for `is true, is false, is unknown ....`
Note that for the Dask SQL usage, we will no longer see any references to IsNotDistinctFrom since that mapping is happening in DataFusion's physical query planner, which we are not using.